### PR TITLE
Add Modal Component

### PR DESCRIPTION
**Pull request by Ongomobile**
Hi Lukas,

Thanks again for your great article and contribution you really helped me a lot.
I hope this Modal Component is useful to others?

Cheers,

Mike Haslam

### CODE REVIEW

File: src/components/Modal.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+// Cutomization props
+const Content = styled.div`
+  margin: ${(props) => props.setMargin || "15% auto"};
+  background-color: ${(props) => props.setBgColor || "white"};
+  border-radius: ${(props) => props.setBorderRadius || "0.25rem"};
+  padding: ${(props) => props.setPadding || "2rem"};
+  position: ${(props) => props.setPosition || "relative"};
+  height: ${(props) => props.setHeight || "50vh"};
+  width: ${(props) => props.setWidth || "50vw"};
```
Comment (gisderdube) [inline src/components/Modal.jsx]: rather remove the `set` prefix

File: src/components/TagsUI.jsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    if (e.keyCode === 8 && hasTitles && tag === "") {
+      e.preventDefault();
+      props.editLastTitle();
+      tagForm.reset();
+    }
+  };
+
+  return (
+    <InputWrapper>
+      <Form ref={(input) => (tagForm = input)}>
```
Comment (gisderdube) [inline src/components/TagsUI.jsx]: not sure if form is the best component, if you don't plan to submit it.

File: src/components/TagsUI.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  return (
+    <TagTitle>
+      <span>{title.text}</span>
+      <CrossIcon src={Cross} onClick={() => props.removeTitle(id)} />
+    </TagTitle>
+  );
+};
+
+const TagInput = (props) => {
+  let text, tagForm;
```
Comment (gisderdube) [inline src/components/TagsUI.jsx]: I guess should use `useRef` here.

File: src/components/TagsUI.jsx [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      props.editLastTitle();
+      tagForm.reset();
+    }
+  };
+
+  return (
+    <InputWrapper>
+      <Form ref={(input) => (tagForm = input)}>
+        <Input
+          ref={(input) => (text = input)}
```
Comment (gisderdube) [inline src/components/TagsUI.jsx]: see above

File: src/components/Modal.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  return (
+    <Overlay>
+      <Content
+        setMargin={props.setMargin}
+        setBgColor={props.setBgColor}
+        setBorderRadius={props.setBorderRadius}
+        setPadding={props.setPadding}
+        setPosition={props.setPosition}
+        setHeight={props.setHeight}
+        setWidth={props.setWidth}
```
Comment (gisderdube) [inline src/components/Modal.jsx]: pass through remaining props, in case the dev wants to override styling or pass any other handlers.

**Review by gisderdube — CHANGES_REQUESTED** [review]

File: src/components/Modal.jsx (hunk 1 above)
Comment (Ongomobile) [reply src/components/Modal.jsx]: Hi Lukas I committed to remove the set prefix

File: src/components/TagsUI.jsx (hunk 2 above)
Comment (Ongomobile) [reply src/components/TagsUI.jsx]: I believe I need the form in the way I used it what would you do differently?

File: src/components/TagsUI.jsx (hunk 3 above)
Comment (Ongomobile) [reply src/components/TagsUI.jsx]: How would you use?

File: src/components/TagsUI.jsx (hunk 4 above)
Comment (Ongomobile) [reply src/components/TagsUI.jsx]: How would you use

File: src/components/Modal.jsx (hunk 5 above)
Comment (Ongomobile) [reply src/components/Modal.jsx]: I committed  passing in the remaining props

File: src/components/TagsUI.jsx (hunk 4 above)
Comment (gisderdube) [reply src/components/TagsUI.jsx]: you can do 
```
  const inputRef = useRef({})
  ...
  <Input ref={inputRef} />
```

See https://reactjs.org/docs/hooks-reference.html#useref for more details

File: src/components/TagsUI.jsx (hunk 3 above)
Comment (gisderdube) [reply src/components/TagsUI.jsx]: As you're not using `onSubmit` or something form related, I think you can simply use a `<div>`.

File: src/components/TagsUI.jsx (hunk 4 above)
Comment (gisderdube) [reply src/components/TagsUI.jsx]: you can then access it with `inputRef.current`

**Review by gisderdube — CHANGES_REQUESTED** [review]

File: src/components/TagsUI.jsx (hunk 4 above)
Comment (Ongomobile) [reply src/components/TagsUI.jsx]: Hi Lukas, sorry I did not have a look at this sooner. I don't get any notifications from GitHub. I'm trying to wrap my head around useRef when I first just change the form to a div as you suggest. it throws an error saying tagForm is not a function. Here is a link to a Code Sandbox for Tags UI https://codesandbox.io/s/ui-styledtags-forked-bef94?

Comment (Ongomobile) [reply src/components/TagsUI.jsx]: Hi Lukas,

 I Updated TagInput to use a controlled input and useState hook

**Pull request closed without merging**